File: 로또의_최고_순위와_최저_순위/로또의_최고_순위와_최저_순위.js
function solution(lottos, win_nums) {
  const {zeroCount, correctCount} = checkCorrectAndZeroCount(lottos, win_nums);
  const maxCorrect = correctCount + zeroCount
  const minCorrect = correctCount

  return [rank(maxCorrect), rank(minCorrect)]
}

const checkCorrectAndZeroCount = (lottos, win_nums) => {
  let zeroCount = 0;
  let correctCount = 0;
  
  for(let lotto of lottos) {
    if(lotto === 0) {
      zeroCount++
      continue
    }
    if(win_nums.includes(lotto)) correctCount++ 
  }

  return {zeroCount, correctCount}
} 

const rank = (score) => {
  if(score > 1) return 7 - score
  else if(score === 0 || score === 1) return 6
}

File: programmers/로또의 최고 순위와 최저 순위/로또의_최고_순위와_최저_순위.py
def solution(lottos, win_nums):
    zeroCount = lottos.count(0)
    correctCount = len(set(lottos) & set(win_nums))
    
    maxCount = correctCount + zeroCount 
    minCount = correctCount
    
    return [7 - max(maxCount, 1), 7 - max(minCount, 1)]